transaction-service/src/test/kotlin/ua/com/foxminded/restClient/NatsControllerTest.kt

```diff
@@ -0,0 +1,137 @@
+
+    val exchangeService = Mockito.mock(CurrencyExchangeService::class.java)
+
+    lateinit var natsUrl: String
+
+    lateinit var natsConnection: Connection
+
+    lateinit var natsController: NatsController
+
+    lateinit var nats: GenericContainer<*>
+
+    @BeforeAll
+    fun setUp() {
+        nats = GenericContainer("nats:latest").withExposedPorts(NATS_PORT, NATS_MGMT_PORT)
+        nats.start()
+        natsUrl = "nats://%s:%d".format(nats.host, nats.getMappedPort(NATS_PORT))
+        natsConnection = Nats.connect(natsUrl)
+        natsController = NatsController(transactionService, exchangeService, natsConnection)
+    }
+
+    @Test
+    fun `nats should return flux transactions when input correct`(){
+        val request: ProtoMessage.TransactionRequestProto = getTransactionRequest()
+        val latch = CountDownLatch(2)
+        val responseMessages: MutableList<Message> = mutableListOf()
+        val expectedSize = 2
+        val dispatcher = natsConnection.createDispatcher {
+            responseMessages.add(it)
+            latch.countDown()
+        }
+        dispatcher.subscribe(NATS_RESPONSE_TOPIC)
+
+        Mockito.`when`(
+            transactionService.findAllByIdAndBetweenDate(
+                any<UUID>(),
+                any<LocalDateTime>(),
+                any<LocalDateTime>(),
+                any<Pageable>()
+            )
+        )
+            .thenReturn(Flux.fromIterable(testDtos))
+        Mockito.`when`(exchangeService.exchangeTo(eq(testDtos[0]), any<Currency>()))
+            .thenReturn(testDtos[0])
+        Mockito.`when`(exchangeService.exchangeTo(eq(testDtos[1]), any<Currency>()))
+            .thenReturn(testDtos[1])
+        natsConnection.publish(NATS_REQUEST_TOPIC, NATS_RESPONSE_TOPIC, request.toByteArray())
+        latch.await()
+        val responseIds = responseMessages.map { ProtoMessage.TransactionResponseProto.parseFrom(it.data).id }
+
+        assertEquals(expectedSize, responseMessages.size)
+        assertEquals(responseIds.contains(testDtos[0].id.toString()), true)
+    }
+
+    @AfterAll
+    fun shutDown() {
+        natsConnection.close()
+        nats.stop()
+    }
+
+    private fun getTransactionRequest(): ProtoMessage.TransactionRequestProto {
+        val startDate = LocalDateTime.parse("2022-11-01T12:00:00").atZone(ZoneId.systemDefault()).toInstant()
+        val endDate = LocalDateTime.parse("2022-11-01T12:00:00").atZone(ZoneId.systemDefault()).toInstant()
+        return ProtoMessage.TransactionRequestProto.newBuilder()
+            .setPersonId("e966f608-4621-11ed-b878-0242ac120002")
+            .setCurrency("UAH")
+            .setStartDate(Timestamp.newBuilder().setSeconds(startDate.epochSecond).setNanos(startDate.nano).build())
+            .setEndDate(Timestamp.newBuilder().setSeconds(endDate.epochSecond).setNanos(endDate.nano).build())
+            .setPage(0)
+            .setSize(2)
+            .build()
```

Comment:
There are some tricks how to make the code simpler:

```suggestion
        return ProtoMessage.TransactionRequestProto.newBuilder().apply {
            personId = "e966f608-4621-11ed-b878-0242ac120002"
            currency = "UAH"
            startDateBuilder.setSeconds(startDate.epochSecond).setNanos(startDate.nano)   
            endDateBuilder.setSeconds(endDate.epochSecond).setNanos(endDate.nano)
            page = 0
            size = 2
         }.build()
```

It's not call to action, just sharing knowledge :)

Reply:
Ok. I just try commit suggestion on GitHub)